### PR TITLE
Enable automatic closing of task with type typeAcceptClose

### PR DESCRIPTION
**Pull request by BartChris**
fixes https://github.com/kitodo/kitodo-production/issues/5300
fixes https://github.com/kitodo/kitodo-production/issues/4693 

I am not quite sure where to redirect if a task is closed after acceptance. I setup a redirect to the desktop for this case.

### CODE REVIEW

**Comment by solth** [comment]
@BartChris thank you for the pull request. I think it would be more appropriate to redirect to the task list instead of the desktop after a task of type "acceptClose" has been closed, but I am not a user of the system. What is your opinion about this, @subhhwendt and @andre-hohmann?

**Review by henning-gerhardt — CHANGES_REQUESTED** [review]
- This change fixes only #5300 but not #4693. If the first task is a normal task and the following task a task of "accept and close" then this task is closed too!
- Returning to desktop after a task is closed in this way feels wrong as I assume that the user is expecting to return back to the task list
- Please provide tests which shows that your change is correct or it is easily happen that this kind of function get broken or missing again.

**Comment by BartChris** [comment]
I implemented a fix for https://github.com/kitodo/kitodo-production/issues/4693 and added a test. The build succeeds, but Codacy fails because a missing Javadoc annotation for the test. I do not know, what i have to change here in order for Codacy to succeed. 

grafik

File: Kitodo/src/test/java/org/kitodo/production/forms/CurrenTaskFormIT.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        Task taskTypeAcceptClose = createAndSaveTask(TaskStatus.OPEN, 1, process, null, true);
+        Task followingTask = createAndSaveTask(TaskStatus.LOCKED, 2, process, null, true);
+        currentTaskForm.setTaskById(taskTypeAcceptClose.getId());
+        currentTaskForm.takeOverTask();
+        Task taskTypeAcceptCloseUpdated = taskService.getById(taskTypeAcceptClose.getId());
+        Task followingTaskUpdated = taskService.getById(followingTask.getId());
+
+        assertEquals("Task of type typeAcceptClose was closed!", TaskStatus.DONE,
+            taskTypeAcceptCloseUpdated.getProcessingStatus());
+        assertEquals("Folowing task is open!", TaskStatus.OPEN, followingTaskUpdated.getProcessingStatus());
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/production/forms/CurrenTaskFormIT.java:78]: Grammar: `Folowing` -> `Following`

File: Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java [hunk 2]
```diff
@@ -652,14 +647,12 @@ private boolean runXPathCondition(Process process, String xpath) throws IOExcept
         return ServiceManager.getProcessService().getNodeListFromMetadataFile(process, xpath).getLength() > 0;
     }
 
-    private void verifyTask(Task task) {
+    private void checkForAutomaticTask(Task task) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java:650]: Maybe there is a better method naming possible like `perfomActionsForAutomaticTask` or something similar which is describing what this method is doing.

**Review by henning-gerhardt — CHANGES_REQUESTED** [review]
Changes looks good, but maybe it is possible to rename the mentioned method and fix the grammar error.

**Comment by henning-gerhardt** [comment]
> The build succeeds, but Codacy fails due to a missing Javadoc annotation for the test. I do not know, what i have to change here in order for Codacy to succeed.

I don't know how this was handled in the past and how @solth is handling this. Easiest fix would be to provide a JavaDoc comment above the test method.

**Comment by solth** [comment]
> Easiest fix would be to provide a JavaDoc comment above the test method.

I agree. Even though we have many tests without JavaDoc in the existing code, we should at least try to have a short JavaDoc for new tests.

File: Kitodo/src/test/java/org/kitodo/production/forms/CurrenTaskFormIT.java (hunk 1 above)
Comment (BartChris) [reply Kitodo/src/test/java/org/kitodo/production/forms/CurrenTaskFormIT.java]: fixed!

File: Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java (hunk 2 above)
Comment (BartChris) [reply Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java]: renamed the method

**Pull request merged**